src/ORM/DataObjectSchema.php

```diff
@@ -688,7 +688,30 @@ public function tableForField($candidateClass, $fieldName)
     }
 
     /**
-     * Returns the class name in the class hierarchy which contains a given
+     * Returns the table name(s) in the class ancestry and descendants which contain a given
+     * field column for a {@link DataObject}. If the field does not exist, this will return null.
+     *
+     * @param string $candidateClass
+     * @param string $fieldName
+     *
+     * @return string
+     */
+    public function tablesForField($candidateClass, $fieldName)
```

Comment:
This duplicates a lot of tableForField, which
 * Only looks at the current class and its ancestors
 * Can more confidently return a single field

Whereas this
 * Looks at the descendants as well as ancestors
 * Returns an array of fields, although quite likely an array of 1

We may want to look at deprecating the original version?

Reply:
Note, DataQuery (not datalist) already had (still has?) logic in it to cover querying of subfields of tables.

This actually caused issues where you had multiple sub-tables that had the same field names as siblings. E.g. `Banner`, `HeaderBanner extends Banner has_one Image`, `FooterBanner extends Banner has_one Image`.

Look up `collidingFields` in DataQuery.

I don't think it's impossible to solve this, but I was frustrated with the complexity and how convoluted the code needed to be to handle these situations.

In order for the ORM to get better it probably needs to be better at managing this complexity by being picky about what problems we want to solve. :) It's purely opinion but I think my `->ofType()` solution is a lot simpler.

Reply:
https://github.com/silverstripe/silverstripe-framework/issues/1683#issuecomment-439180349

Reply:
> Note, DataQuery (not datalist) already had (still has?) logic in it to cover querying of subfields of tables.

True, I might look at refactoring it into here.

> It's purely opinion but I think my ->ofType() solution is a lot simpler.

Simpler for us, harder for users.

Reply:
> This actually caused issues where you had multiple sub-tables that had the same field names as siblings.

That's why it's tablesForField() and not tableForField(). ;) Thankfully(?) our test suite actually contains these cases so I identified it relatively early. This PR hands such changes.

Although, now that I mention it, my test cases were only for colliding _fields_ not colliding _relations_.

IMO we should probably throw a warning/deprecation when you're using this annoying capability, but that's a separate ticket.